lib/api/core/entrypoints/embedded/protocols/websocket.js

```diff
@@ -0,0 +1,308 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2017 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const
+  debug = require('../../../../../kuzzleDebug')('kuzzle:entry-point:protocols:websocket'),
+  Protocol = require('./protocol'),
+  Request = require('kuzzle-common-objects').Request,
+  WebSocketServer = require('uws').Server,
+  ClientConnection = require('../clientConnection'),
+  {
+    BadRequestError,
+    InternalError: KuzzleInternalError,
+    SizeLimitError
+  } = require('kuzzle-common-objects').errors;
+
+/**
+ * @class WebsocketProtocol
+ */
+class WebsocketProtocol extends Protocol {
+  constructor () {
+    super();
+
+    this.channels = {};
+    this.connectionPool = {};
+    this.server = null;
+    this.protocol = 'websocket';
```

Comment:
nitpicking: here we have a class property `this.protocol`, while in [socketio](https://github.com/kuzzleio/kuzzle/blob/no-proxy/lib/api/core/entrypoints/embedded/protocols/socketio.js#L25), we use a module variable `_protocol`.
This should be harmonized.